### PR TITLE
fixed container bindings use of container

### PR DESCRIPTION
**Pull request by luceos**
> No time to write the issue, someone 🙏 There's supposed to be an issue somewhere for this.

**Fixes #0000**

**Changes proposed in this pull request:**

**Reviewers should focus on:**

**Screenshot**

**Confirmed**

- [ ] Frontend changes: tested on a local Flarum installation.
- [ ] Backend changes: tests are green (run `composer test`).

**Required changes:**

- [ ] Related documentation PR: (Remove if irrelevant)
- [ ] Related core extension PRs: (Remove if irrelevant)

### CODE REVIEW

File: src/User/UserServiceProvider.php [hunk 1]
```diff
@@ -90,18 +92,16 @@ protected function registerPasswordCheckers()
...
 
-        User::setHasher($this->container->make('hash'));
-        User::setPasswordCheckers($this->container->make('flarum.user.password_checkers'));
-        User::setGate($this->container->makeWith(Access\Gate::class, ['policyClasses' => $this->container->make('flarum.policies')]));
-        User::setDisplayNameDriver($this->container->make('flarum.user.display_name.driver'));
-
-        $events = $this->container->make('events');
+        User::setHasher($container->make('hash'));
+        User::setPasswordCheckers($container->make('flarum.user.password_checkers'));
+        User::setGate($container->makeWith(Access\Gate::class, ['policyClasses' => $this->container->make('flarum.policies')]));
```
Comment (askvortsov1) [inline src/User/UserServiceProvider.php:103]: Missed one

**Review by askvortsov1 — CHANGES_REQUESTED** [review]
Found one you missed

File: src/User/UserServiceProvider.php (hunk 1 above)
Comment (askvortsov1) [inline src/User/UserServiceProvider.php:103]:
```suggestion
        User::setGate($container->makeWith(Access\Gate::class, ['policyClasses' => $this->container->make('flarum.policies')]));
```
```suggestion
        User::setGate($container->makeWith(Access\Gate::class, ['policyClasses' => $container->make('flarum.policies')]));
```

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

**Review by askvortsov1 — APPROVED** [review]

**Review by tankerkiller125 — APPROVED** [review]

**Comment by askvortsov1** [comment]
PArt of https://github.com/flarum/core/issues/2763

**Pull request merged**